app/containers/AddressChooserModalContainer.tsx

```diff
@@ -0,0 +1,170 @@
+    });
+
+    this.handleShowPreviousAddresses = this.handleShowPreviousAddresses.bind(this);
+    this.handleShowNextAddresses = this.handleShowNextAddresses.bind(this);
+    this.handleAddressChosen = this.handleAddressChosen.bind(this);
+    this.handleDerivationPathChange = debounce(this.handleDerivationPathChange.bind(this), 300);
+  }
+
+  public componentDidUpdate(
+    _prevProps: IAddressChooserModalContainerProps,
+    prevState: IAddressChooserModalContainerState
+  ) {
+    if (
+      this.state.derivationPath !== prevState.derivationPath ||
+      this.state.startingIndex !== prevState.startingIndex
+    ) {
+      this.obtainData().catch(error => {
+        throw new Error(error);
+      });
+    }
+  }
+
+  public render() {
+    return (
+      <AddressChooserModalComponent
+        loading={this.state.loading}
+        derivationPath={this.state.derivationPath}
+        addresses={this.state.addresses}
+        previousAddressesDisabled={this.state.startingIndex <= 0}
+        handleShowPreviousAddresses={this.handleShowPreviousAddresses}
+        handleShowNextAddresses={this.handleShowNextAddresses}
+        handleAddressChosen={this.handleAddressChosen}
+        handleDerivationPathChange={this.handleDerivationPathChange}
+      />
+    );
+  }
+
+  private async obtainData() {
+    this.setState({ loading: true });
+    const stateCopy = { ...this.state };
+    stateCopy.addresses = {};
+    stateCopy.loading = false;
+
+    for (
+      let i = stateCopy.startingIndex;
+      i < stateCopy.startingIndex + NUMBER_OF_ADDRESSES;
+      i = i + 1
+    ) {
+      stateCopy.addresses[stateCopy.derivationPath + i] = {
+        address: null,
+        ETH: null,
+      };
+    }
+
+    let addresses = null;
+    try {
+      addresses = await ledgerInstance.getMultipleAccountsAsync(
+        stateCopy.derivationPath,
+        stateCopy.startingIndex,
+        Object.keys(stateCopy.addresses).length
+      );
+    } catch (error) {
+      throw new Error(error);
+    }
+
+    try {
+      for (const dp in addresses) {
+        if (addresses.hasOwnProperty(dp)) {
+          const address = addresses[dp];
+          stateCopy.addresses[dp].address = address;
+          stateCopy.addresses[dp].ETH = web3Instance.fromWei(
+            await web3Instance.eth.getBalanceAsync(address),
+            "ether"
+          );
+
+          /* async version just in case we need it
+          web3Instance.eth.getBalanceAsync(address).then(
+            function(dep: string) {
+              return (balanceInWei: BigNumber) => {
+                const oldState = this.state;
+                oldState.addresses[dep].ETH = web3Instance.fromWei(balanceInWei, "ether");
+                this.setState(oldState);
+              };
+            }.bind(this)(dp)
+          );*/
+        }
+      }
+    } catch (error) {
+      throw new Error(error);
+    }
+
+    this.setState(stateCopy);
+  }
+
+  private handleShowPreviousAddresses() {
```

Comment:
use `private handleShowPreviousAddresses  = () => {` to get `this` bound to correct context

Reply:
same aplies to rest of bound methods